virttest/utils_netperf.py

```diff
@@ -158,26 +158,91 @@ def env_cleanup(self, clean_all=True):
+        try:
+            max_cpus_fp, res_cpus = self.session.cmd_output(
+                'grep -orP "^#define *MAXCPUS *\K[0-9]+" %s' % self.netperf_dir
+            ).split(":", maxsplit=1)
+            max_cpus = int(res_cpus)
+        except aexpect.ShellError as err:
+            raise NetperfPackageError(
+                "Failed to find MAXCPUS value and file under '%s', see: %s"
+                % (self.netperf_dir, err)
+            )
+        except ValueError:
+            raise NetperfPackageError(
+                "Invalid MAXCPUS value found in file '%s': %s" % (max_cpus_fp, res_cpus)
+            )
+        LOG.debug(
+            "Found MAXCPUS is defined in %s with value %d" % (max_cpus_fp, max_cpus)
+        )
+        return max_cpus_fp, max_cpus
+
+    def _mod_max_cpus(self):
+        """
+        netperf is hardcoded to support up to 256 CPUS.
+        Modify the value from the source code previous to compilation to the
+        desired value.
+        """
+        # Obtain host/guest/vm/target netperf running sytem's number of CPUs
+        try:
+            n_cpus = int(
+                self.session.cmd_output('lscpu | grep -oP "^CPU\(s\)\: *\K[0-9]+"')
+            )
+            LOG.debug("Found %d cpus in the system running netperf" % n_cpus)
+        except aexpect.ShellError as err:
+            LOG.warning(
+                "Couldn't retrieve the number of cpus in the system: see %s" % err
+            )
+            n_cpus = 0
+        except ValueError:
+            LOG.warning("Couldn't parse the number of cpus in the system")
+            n_cpus = 0
+        max_cpus_file_path, current_max_cpus = self._get_current_max_cpus()
+        if current_max_cpus >= n_cpus:
+            LOG.debug("Bypassing netperf's MAXCPUS value modification")
+            return
+        LOG.info("Increasing netperf's MAXCPUS to %d" % n_cpus)
+        sed_cmd = 'sed -i "s/^\(#define *MAXCPUS *\)[0-9]* /\\1%d/g" %s' % (
+            n_cpus,
+            max_cpus_file_path,
+        )
+        self.session.cmd_output(sed_cmd, timeout=5)
+
     def pack_compile(self, compile_option=""):
         pre_setup_cmd = "cd %s " % self.netperf_base_dir
         pre_setup_cmd += " && %s %s" % (self.decomp_cmd, self.netperf_file)
-        pre_setup_cmd += " && cd %s " % self.netperf_dir
         # Create dict to make other OS architectures easy to extend
         build_type = {"aarch64": "aarch64-unknown-linux-gnu"}
         build_arch = self.session.cmd_output("arch", timeout=60).strip()
+        self.env_cleanup(clean_all=False)
+        try:
+            self.session.cmd("echo $?", timeout=60)
+            self.session.cmd(pre_setup_cmd, timeout=600)
+        except aexpect.ShellError as e:
+            raise NetperfPackageError("Compilation setup failed: %s" % e)
+        try:
+            self._mod_max_cpus()
+        except aexpect.ShellError as err:
+            raise NetperfPackageError("Max CPU modification failed: %s" % err)
```

Comment:
```suggestion
            LOG.error("Max CPU modification failed: %s" % err)
```
Hi @bgartzi 
I have a question about the `pack_compile` procedure. I've noticed that the test case will fail if it's unable to modify the max_cpus setting.

Is this CPU configuration a mandatory requirement for the tool to run correctly? I want to confirm because this step could be blocking valid test runs on platforms where this setting isn't critical.

If the feature owners confirm it's a required step, that's fine. I just want to ensure we aren't causing unnecessary failures.

What are your thoughts on this? Let me know if you see any issues with this approach.
Thanks.
CC @rh-jugraham @yanglei-rh @yiqianwei @fbq815 @zhenyzha

Reply:
Hi @bgartzi 
Thanks for the explanation—your solution works for me.
My only consideration is that calling `_get_current_max_cpus` could introduce new NetperfPackageError exceptions, though this is likely a rare edge case.

For example:
```
raise NetperfPackageError(
    "Failed to find MAXCPUS value and file under '%s', see: %s"
    % (self.netperf_dir, err)
)
```
and 
 ```
 raise NetperfPackageError(
      "Invalid MAXCPUS value found in file '%s': %s" % (max_cpus_fp, res_cpus)
  )
```

BTW, since the original issue was on ARM, I'm asking @fbq815 to review this change from the perspective of the s390x platform.
@fbq815, could you please take a look?
Thanks.